storage/innobase/include/fil0fil.h

```diff
@@ -1619,13 +1619,15 @@ and a suffix. The string must be freed by caller with ut_free().
 @param[in] path NULL or the directory path or the full path and filename.
 @param[in] name NULL if path is full, or Table/Tablespace name
 @param[in] suffix NULL or the file extention to use.
+@param[in] dir_path true, if the intended path name is directory.
 @return own: file name */
 char*
 fil_make_filepath(
 	const char*	path,
 	const char*	name,
 	ib_extention	suffix,
-	bool		strip_name);
+	bool		strip_name,
+	bool		dir_path = false);
```

Comment:
Instead of adding a new Boolean parameter, could we add a `suffix` enumeration value to say that it is a directory, similar to the value `NO_EXT` that is only being used in `SysTablespace::parse_params()`. We might also refactor that code and repurpose `NO_EXT`. In that case, `Datafile::make_filepath()` would have to assert that it is not passing this `suffix` value to `fil_make_filepath()`.

By the way, the parameter `strip_name` is not documented in the function comment.

Reply:
We don't want any suffix and using NO_EXT should be good. Adding another blank suffix of different name would be confusing I think.

Actually the function fil_make_filepath() works perfectly with the current set of inputs. The only issue is the debug assert 
```
/* If we are going to strip a name off the path, there better be a
path and a new name to put back on */
ut_ad(!trim_name || (path != NULL && name != NULL))
```
The new boolean parameter is added to preserve the debug assert for all other cases.
`bool		dir_path = false); `
and relaxing the assert when we are creating the directory path. So, I think we should not mix it with the functional parameters that are passed to drive the behaviour of the function.

If we want to simplify (removing the bool parameter) I could see 2 viable approaches.

1. Comment the debug assert that prevent the function to return a directory in debug mode.

OR

2. Have a wrapper function
```
fil_make_file_path(...) {
   Debug Assert
   fil_make_file_path_low()
} 
fil_make_directory_path() {
   fil_make_file_path_low()
}
Rename fil_make_file_path to fil_make_file_path_low and make it static scope.
```
I will upload 2 commits one with each method respectively. Please let me know which one you like. Or you still prefer yet any different approach.

Reply:
I think that having two wrapper functions (`inline` ones) would be best. In this way, there should be no overhead in `CMAKE_BUILD_TYPE=RelWithDebInfo` builds.

Reply:
Thanks, updated the patch.